lib/base/benchmark.hpp

```diff
@@ -0,0 +1,37 @@
+/* Icinga 2 | (c) 2024 Icinga GmbH | GPLv2+ */
+
+#pragma once
+
+#include "base/atomic.hpp"
+#include <chrono>
+
+namespace icinga
+{
+
+/**
+ * Benchmark result.
+ *
+ * @ingroup base
+ */
+class Benchmark
+{
+public:
+	using Clock = std::chrono::steady_clock;
+
+	Benchmark& operator+=(const Clock::duration&) noexcept;
+	Benchmark& operator+=(const Clock::time_point&) noexcept;
+
+	/**
+	 * @return The total accumulated time in seconds
+	 */
+	template<class T>
+	operator T() const noexcept
+	{
+		return std::chrono::duration<T>(Clock::duration(m_Sum.load(std::memory_order_relaxed))).count();
+	}
```

Comment:
> The total accumulated time *in seconds*

Isn't this templated because `T` can be used to specify the unit, i.e. it could be something other than seconds?

Reply:
No, T is just [an arithmetic type, or a class emulating an arithmetic type, representing the number of ticks](https://en.cppreference.com/w/cpp/chrono/duration).